### PR TITLE
Adds validation that the service account specified in 'runAsUser' has …

### PR DESCRIPTION
**Pull request by ttomsu**
…WRITE permission to application

@jtk54 @duftler @cfieber @ajordens PTAL.

### CODE REVIEW

File: front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  @Unroll
+  def "should validate run as user access"() {
+    setup:
+    FiatPermissionEvaluator evaluator = Mock(FiatPermissionEvaluator)
+    @Subject AuthorizationSupport authorizationSupport = new AuthorizationSupport(permissionEvaluator: evaluator)
+
+    when:
+    Pipeline p = new Pipeline()
+    authorizationSupport.hasRunAsUserPermission(p)
```
Comment (ajordens) [inline front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy]: Are you expecting that `authorizationSupport.hasRunAsUserPermission(p)` asserts true?

Normally that should be done in `then:`.

```
when:
emptyPipelineHasRunAsUserPermission = hasRunAsUserPermission(new Pipeline()) 

then:
emptyPipelineHasRunAsUserPermission == true
```

or

```
expect:
authorizationSupport.hasRunAsUserPermission(new Pipeline()) == true
authorizationSupport.hasRunAsUserPermission(new Pipeline(triggers: [])) == true
```

Am I miss interpreting something here?

File: front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import spock.lang.Subject
+import spock.lang.Unroll
+
+public class AuthorizationSupportSpec extends Specification {
+
+  @Unroll
+  def "should validate run as user access"() {
+    setup:
+    FiatPermissionEvaluator evaluator = Mock(FiatPermissionEvaluator)
+    @Subject AuthorizationSupport authorizationSupport = new AuthorizationSupport(permissionEvaluator: evaluator)
```
Comment (ajordens) [inline front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy]: `@Subject` is probably a useless annotation to apply within a `setup`.

could be:

```
public class AuthorizationSupportSpec extends Specification {
  def evaluator = Mock(FiatPermissionEvaluator)

  @Subject
  def authorizationSupport = new AuthorizationSupport(permissionEvaluator: evaluator)

  @Unroll
  def "should validate run as user access"() {
     when:
     ...

    then:
    ...
}
```

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupport.groovy [hunk 3]
```diff
@@ -31,15 +32,34 @@ class AuthorizationSupport {
 
   boolean hasRunAsUserPermission(Pipeline pipeline) {
     List<String> runAsUsers = pipeline.triggers*.runAsUser
```
Comment (ajordens) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupport.groovy]: `pipeline.triggers.findResults { it.runAsUser }` should avoid the need to `.removeAll([null])`

File: front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy (hunk 1 above)
Comment (ttomsu) [inline front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy]: This cleaned up the code nicely. Good suggestions.

File: front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy (hunk 2 above)
Comment (ttomsu) [inline front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupportSpec.groovy]: Done.

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupport.groovy (hunk 3 above)
Comment (ttomsu) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/AuthorizationSupport.groovy]: Good call here, too. Thanks!

**Comment by ajordens** [comment]
LGTM, thanks @ttomsu!

**Pull request merged**